lightkurve/prf/scenemodels.py

```diff
@@ -0,0 +1,670 @@
+"""Provides tools to model a Kepler image."""
+from __future__ import division, print_function
+
+import logging
+from matplotlib import pyplot as plt
+import numpy as np
+from tqdm import tqdm
+
+from oktopus import Prior, GaussianPrior, UniformPrior, PoissonPosterior
+
+from .prfmodels import KeplerPRF
+from ..utils import plot_image
+
+
+__all__ = ['GaussianPrior', 'UniformPrior', 'FixedValuePrior',
+           'StarPrior', 'BackgroundPrior', 'FocusPrior', 'MotionPrior',
+           'StarParameters', 'BackgroundParameters', 'FocusParameters',
+           'MotionParameters', 'SceneModelParameters',
+           'SceneModel', 'PRFPhotometry']
+
+
+log = logging.getLogger(__name__)
+
+
+class FixedValuePrior(Prior):
+    """An improper prior with a likelihood probability of 1 at a single fixed
+    value and -inf elsewhere. This is similar to a Dirac Delta function,
+    except this function does not peak at infinity so that it can be used
+    in numerical optimization functions.  It does not integrate to one as a
+    result and is therefore an "improper distribution".
+
+    Attributes
+    ----------
+    value : int or array-like of ints
+        The fixed value.
+
+    Examples
+    --------
+    >>> fp = FixedValuePrior(1)
+    >>> fp(1)
+    -0.0
+    >>> fp(0.5)
+    inf
+    """
+    def __init__(self, value, name=None):
+        self.value = np.asarray([value]).reshape(-1)
+        self.name = name
+
+    def __repr__(self):
+        return "<FixedValuePrior(value={})>".format(self.value)
+
+    @property
+    def mean(self):
+        """Returns the fixed value."""
+        return self.value
+
+    @property
+    def variance(self):
+        """Returns zero."""
+        return 0
+
+    def evaluate(self, params):
+        """Returns the negative log pdf."""
+        if self.value == params:
+            return -0.0
+        return np.inf
+
+    def gradient(self, params):
+        if self.value == params:
+            return 0.0
+        return np.inf
```

Comment:
I'm not sure about that

Reply:
The problem is that the gradient of ``UniformPrior`` is undefined at the boundaries. For ``FixedValuePrior`` we don't even have boundaries. My point is that if a parameter is fixed we shouldn't try to compute ``d model / d param``

Reply:
Ok!  I'm going to skip uncertainties for now and raise a `NotImplementedError` for `FixedValuePrior.gradient()`.